src/heitech.zer0mqXt.core/Patterns/Socket.cs

```diff
@@ -78,71 +82,119 @@ public Socket(SocketConfiguration configuration)
-                            TResult result = factory(actualRequestResult.IsSuccess ? actualRequestResult.GetResult() : new T());
-
-                            response = new RequestReplyMessage<TResult>(_configuration, result, actualRequestResult.IsSuccess);
-
-                        }
-                        catch (System.Exception ex)
-                        {
-                            // failure to parse or any other exception leads to a non successful response, which then in turn can be handled on the request side
-                            _configuration.Logger.Log(new ErrorLogMsg($"Responding to [Request:{typeof(T)}] with [Response:{typeof(TResult)}] did fail: " + ex.Message));
-                            response = new RequestReplyMessage<TResult>(_configuration, default(TResult), success: false);
-                        }
-
-                        // try send response with timeout
-                        bool noTimeout = rsSocket.TrySendMultipartMessage(_configuration.TimeOut, response);
-                        if (!noTimeout)
-                            _configuration.Logger.Log(new ErrorLogMsg($"Responding to [Request:{typeof(T)}] with [Response:{typeof(TResult)}] timed-out after {_configuration.TimeOut}"));
-
-                        if (respondOnce || cancellationToken.IsCancellationRequested)
-                            runReponder = false;
-                    }
+                    // poller blocks, so it has to be started after the eventhandle is set
+                    poller.RunAsync();
+                    
+                    // open resetevent after binding to the socket and when the poller is started
+                    eventHandle.Set();
                 }
-                catch (NetMQ.TerminatingException terminating)
+                catch (Exception exception)
                 {
-                    _configuration.Logger.Log(new ErrorLogMsg(terminating.Message));
+                    _configuration.Logger.Log(new ErrorLogMsg(exception.Message));
+                    Dispose();
                 }
-            }, cancellationToken == default ? CancellationToken.None : cancellationToken);
-            // wait for the Set inside the background thread
+            }, cancellationToken);
+
+            // wait for the Set inside the background thread so we can know at the calling client that the server is set up properly
             eventHandle.WaitOne();
         }
 
+
+        private void ResponseHandlerCallback<T, TResult>(ResponseSocket socket, Func<T, TResult> factory, CancellationToken token)
+            where T : class, new()
+            where TResult : class
+        {
+            try
+            {
+                if (!respondingIsActive || token.IsCancellationRequested)
+                {
+                    Dispose();
+                    _configuration.Logger.Log(new InfoLogMsg("Task was cancelled by cancellationRequest"));
+                    return;
+                }
```

Comment:
Could we move this if-block before the try-block? I don't see how this could throw an `NetMQ.TerminatingException`.